libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt

```diff
@@ -0,0 +1,768 @@
+         */
+        fun clientAppName(clientAppName: String): Builder {
+            directionsBuilder.clientAppName(clientAppName)
+            return this
+        }
+
+        /**
+         * Required to call when this is being built. If no access token provided,
+         * [ServicesException] will be thrown.
+         *
+         * @param accessToken Mapbox access token, You must have a Mapbox account inorder to use
+         * the Optimization API
+         * @return this builder for chaining options together
+         * @since 0.5.0
+         */
+        fun accessToken(accessToken: String): Builder {
+            directionsBuilder.accessToken(accessToken)
+            return this
+        }
+
+        /**
+         * Optionally change the APIs base URL to something other then the default Mapbox one.
+         *
+         * @param baseUrl base url used as end point
+         * @return this builder for chaining options together
+         * @since 0.5.0
+         */
+        fun baseUrl(baseUrl: String): Builder {
+            directionsBuilder.baseUrl(baseUrl)
+            return this
+        }
+
+        /**
+         * Indicates from which side of the road to approach a waypoint.
+         * Accepts <tt>unrestricted</tt> (default), <tt>curb</tt> or <tt>null</tt>.
+         * If set to <tt>unrestricted</tt>, the route can approach waypoints
+         * from either side of the road. If set to <tt>curb</tt>, the route will be returned
+         * so that on arrival, the waypoint will be found on the side that corresponds with the
+         * <tt>driving_side</tt> of the region in which the returned route is located.
+         * If provided, the list of approaches must be the same length as the list of waypoints.
+         *
+         * @param approaches null if you'd like the default approaches,
+         * else one of the options found in
+         * [com.mapbox.api.directions.v5.DirectionsCriteria.ApproachesCriteria].
+         * @return this builder for chaining options together
+         * @since 0.15.0
+         */
+        fun addApproaches(vararg approaches: String): Builder {
+            directionsBuilder.addApproaches(*approaches)
+            return this
+        }
+
+        /**
+         * Optionally, set which input coordinates should be treated as waypoints / separate legs.
+         * Note: coordinate indices not added here act as silent waypoints
+         *
+         *
+         * Most useful in combination with <tt>steps=true</tt> and requests based on traces
+         * with high sample rates. Can be an index corresponding to any of the input coordinates,
+         * but must contain the first ( 0 ) and last coordinates' index separated by <tt>;</tt>.
+         *
+         * @param indices integer array of coordinate indices to be used as waypoints
+         * @return this builder for chaining options together
+         */
+        fun addWaypointIndices(@IntRange(from = 0) vararg indices: Int): Builder {
```

Comment:
Sould it be nullable? Cause in Java code it was as:
`public Builder addWaypointIndices(**@Nullable** @IntRange(from = 0) Integer... indices) {`

Reply:
Yeah `waypointIndices` is an optional parameter.